File: Python/Endpoint Examples/JSON Payload/pdf-with-page-boxes-set.py
import requests
import json

with open('/path/to/file', 'rb') as f:
    upload_data = f.read()

print("Uploading file...")
upload_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_data,
                    headers={'Content-Type': 'application/octet-stream', 'Content-Filename': 'file.pdf', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Upload response status code: " + str(upload_response.status_code))

if upload_response.ok:
    upload_response_json = upload_response.json()
    print(json.dumps(upload_response_json, indent = 2))


    uploaded_id = upload_response_json['files'][0]['id']
    box_options = {
        "boxes": [
            {
                "box": "media",
                "pages": [
                    {
                        "range": "1",
                        "left": 100,
                        "top": 100,
                        "bottom": 100,
                        "right": 100
                    }
                ]
            }
        ]
    }
    boxes_data = { "id" : uploaded_id, "boxes": json.dumps(box_options) }

    print(json.dumps(boxes_data, indent = 2))


    print("Processing file...")
    boxes_response = requests.post(url='https://api.pdfrest.com/pdf-with-page-boxes-set',
                        data=json.dumps(boxes_data),
                        headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



    print("Processing response status code: " + str(boxes_response.status_code))
    if boxes_response.ok:
        boxes_response_json = boxes_response.json()
        print(json.dumps(boxes_response_json, indent = 2))

    else:
        print(boxes_response.text)
else:
    print(upload_response.text)

File: JavaScript/Endpoint Examples/JSON Payload/pdf-with-page-boxes-set.js
var axios = require("axios");
var FormData = require("form-data");
var fs = require("fs");

// By default, we use the US-based API service. This is the primary endpoint for global use.
var apiUrl = "https://api.pdfrest.com";

/* For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
 * For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 */
//var apiUrl = "https://eu-api.pdfrest.com";

var upload_data = fs.createReadStream("/path/to/file");

var upload_config = {
  method: "post",
  maxBodyLength: Infinity,
  url: apiUrl + "/upload",
  headers: {
    "Api-Key": "xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
    "Content-Filename": "filename.pdf",
    "Content-Type": "application/octet-stream",
  },
  data: upload_data, // set the data to be sent with the request
};

// send request and handle response or error
axios(upload_config)
  .then(function (upload_response) {
    console.log(JSON.stringify(upload_response.data));
    var uploaded_id = upload_response.data.files[0].id;

    const boxOptions = {
      boxes: [
        {
          box: "media",
          pages: [
            {
              range: "1",
              left: 100,
              top: 100,
              bottom: 100,
              right: 100,
            },
          ],
        },
      ],
    };

    var boxes_config = {
      method: "post",
      maxBodyLength: Infinity,
      url: apiUrl + "/pdf-with-page-boxes-set",
      headers: {
        "Api-Key": "xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
        "Content-Type": "application/json",
      },
      data: {
        id: uploaded_id,
        boxes: JSON.stringify(boxOptions),
      }, // set the data to be sent with the request
    };

    // send request and handle response or error
    axios(boxes_config)
      .then(function (boxes_response) {
        console.log(JSON.stringify(boxes_response.data));
      })
      .catch(function (error) {
        console.log(error);
      });
  })
  .catch(function (error) {
    console.log(error);
  });
